sdks/python/apache_beam/io/gcp/bigquery_tools.py

```diff
@@ -101,6 +101,10 @@
 # Timeout for a BQ streaming insert RPC. Set to a maximum of 2 minutes.
 BQ_STREAMING_INSERT_TIMEOUT_SEC = 120
 
+_PROJECT_PATTERN = r'[a-z][a-z0-9-]{4,28}[a-z0-9]'
```

Comment:
> Must be 6 to 30 characters in length.

I missed that the regex was supposed to allow project_id to be optional, running again with that change. In this case, we check `{4,28}` because the first and last characters have different patterns (must begin with a letter and can't end with a hyphen).